client/src/model-source/diagram-server.ts

```diff
@@ -132,7 +135,7 @@ export abstract class DiagramServer extends ModelSource {
             case ServerStatusAction.KIND:
                 return this.handleServerStateAction(action as ServerStatusAction)
         }
-        return true
+        return !(action as any)[receivedFromServerProperty]
```

Comment:
Shouldn't this even be checked at the beginning of `handleLocally`?

Reply:
I don't think so. There are actions from the server that are directly handled in the DS (and maybe in other action handlers as well). One example is the ServerStatusAction. We could already process such actions in the `messageReceived`, but that would introduce yet another contract.